packages/php-wasm/compile/php/php_wasm.c

```diff
@@ -3,6 +3,38 @@
  *
  * This file abstracts the entire PHP API with the minimal set
  * of functions required to run PHP code from JavaScript.
+ *
+ * !! NOTE FOR FUTURE MAINTAINERS !!
+ *
+ * When debugging memory errors, it's important to understand the behavior of the stack in the Emscripten environment:
+ *
+ * - The stack does not grow automatically. If the stack runs out of space, it will not be expanded automatically.
+ * - The stack size is limited. The default stack size in Emscripten is considerably smaller than in a typical C environment. There have been discussions about reducing it even further.
+ *
+ * When passing in long strings from JS, Use the heap instead of the stack to hold large chunks of data.
+ * You can see an example of how to do this below.
+ *
+ * // Set the memory
+ * const size = Module.lengthBytesUTF8(body);
+ * const addr = Module._malloc(size + 1);
+ * Module.stringToUTF8(body, addr, size + 1);
+ *
+ * // Call your function
+ * Module.ccall('wasm_do_something', null, NUMBER], [addr]);
+ *
+ * // Clean up the memory. REQUIRED if the C code doesn't do this!!!
+ * Module._free(addr);
+ *
+ * !! NOTE !!
+ *
+ * If `lengthBytesUTF8` & `stringToUTF8` encounter a split surrogate at the end of a string,
+ * they will fill in the missing bytes with the first valid code point in that range.
```

Comment:
I'm not sure what we're trying to communicate here, but it doesn't seem accurate. We could clarify this entire comment and move it to above where the sample code is, and based on your experiment, prune it down to something really basic.

```c
/*
 * …
 * // Set the Memory
 * // Note that if provided a string which cannot be converted to UTF-8
 * // then the length reported and the bytes converted will match, and
 * // the transferred string will have the replacement character U+FFFD
 * // wherever the invalid code units appeared in the JavaScript string.
 * // @see {link to emscripten page documenting functions}
 * // Transferring a non-string body is not yet supported.
 */ 
```